Sources/Purchasing/VerificationResult.swift

```diff
@@ -69,3 +69,29 @@ extension VerificationResult: DefaultValueProvider {
     static let defaultValue: Self = .notVerified
 
 }
+
+extension VerificationResult {
+
+    /// - Returns: the most restrictive ``VerificationResult`` based on the cached verification and
+    /// the response verification.
+    static func from(cache cachedResult: Self, response responseResult: Self) -> Self {
+        switch (cachedResult, responseResult) {
+        case (.notVerified, .notVerified),
+            (.verified, .verified),
+            (.failed, .failed):
+            return cachedResult
+
+        // These shouldn't happen because `ETagManager` will ignore not verified cached responses
+        // if verification is enabled
+        case (.notVerified, .verified): return .notVerified
+        case (.notVerified, .failed): return .failed
+
+        case (.verified, .notVerified): return .notVerified
+        case (.verified, .failed): return .failed
+
+        case (.failed, .notVerified): return .failed
```

Comment:
Perhaps we should just store wether the validation of the request was ok/bad and map that to the whole verified/notverified/failed options on the fly given the config setting.

It might simplify the logic about the signature validation in a few places...

Thus, changes in enforcement do not affect the cached data. The only path to care will be in case of enforcing and cached data is BAD, which is is initial upgrade path we have already considered (drop cache, depend on network retrieval)

Reply:
Responses that fail validation are never cached though (that's tested too).

Reply:
Ah right, so these cases should never happen either. Maybe we should make it clearer grouping all the cases that should never happen. The current comment above makes me think the only cases shouldn't happen are the `notVerified` cached ones.

Reply:
Done.